doc/plugin_server_upstreamca_awscertificatemanager.md

```diff
@@ -0,0 +1,53 @@
+# Server plugin: UpstreamCA "awscertificatemanager"
+
+The `awscertificatemanager` plugin uses a certificate authority from AWS
+Certificate Manager (ACM) Private Certificate Authority (PCA) to sign
+intermediate signing certificates for SPIRE Server.
+
+The plugin accepts the following configuration options:
+
+| Configuration           | Description                                            |
+| ----------------------- | --------------------------------------------           |
+| region                  | AWS Region to use                                      |
+| cert_autority_arn       | ARN of the "upstream" CA certificate                   |
+| ca_signing_template_arn | ARN of the signing template to use for the server's CA |
+| signing_algorithm       | Signing algorithm to use for the server's CA           |
```

Comment:
> Defaulting is possible for sure, but it also depends on the key type used for the CA stored in ACM.

Ah yes, good point

> If we assume RSA is the standard key type, SHA256WITHRSA could be a decent trade between security and verification performance as a default, but other opinions are certainly welcomed here.

How would you feel about selecting either SHA256WITHRSA or SHA256WITHECDSA based on the key type of the configured PCA? Since we are doing a Describe when we boot up, looks like we can pull the CA config from there (and subsequently the key algorithm). If this is a Bad Idea, I'd probably prefer no default over one that sometimes fails.

(on this note, we could probably also detect misconfigured signing algorithms, but I wouldn't block this PR for that)

Reply:
> Additionally, I think it could make sense to make arn:aws:acm-pca:::template/SubordinateCACertificate_PathLen0/V1 the default CA signing template, as I believe that would be most commonly used.

No idea what would be normal, but if you think this is a good default then let's do it!

Reply:
Looking again at the describe call response, I remembered that a preferred signing algorithm is set on CA creation (and returned in describe), so I'll fall back to that as the default.

My only hesitance against validating the signing algorithm from the key type is we would need to maintain an enumeration of supported key types, and this is bound to change.

Reply:
> Looking again at the describe call response, I remembered that a preferred signing algorithm is set on CA creation (and returned in describe), so I'll fall back to that as the default.

Perfect :) does AWS select a sensible default?

> My only hesitance against validating the signing algorithm from the key type is we would need to maintain an enumeration of supported key types, and this is bound to change.

Totally... I think we'll have to deal w/ this over time anyway if we're going to document the options etc... but if the AWS preference has defaults and we can use those, then that sounds like our best bet. Thanks for the thought around this, these little details go a long way in the overall usability story

Reply:
AWS selects the defaults we were discussing in fact (SHA256 with appropriate algorithm). Added defaults for both fields.